packages/flutter/lib/src/rendering/sliver_app_bar.dart

```diff
@@ -210,6 +210,7 @@ abstract class RenderSliverScrollingAppBar extends RenderSliverAppBar {
       scrollExtent: maxExtent,
       paintExtent: paintExtent.clamp(0.0, constraints.remainingPaintExtent),
       maxPaintExtent: maxExtent,
+      hasVisualOverflow: true,
```

Comment:
these should each have a comment saying that we're being extra-conservative here to trade the cost of calculating whether we have overflow for the cost of always doing clipping.